homestar-runtime/src/cli/init.rs

```diff
@@ -0,0 +1,372 @@
+                        Settings::path().join("homestar.pem")
+                    };
+
+                    let path = Text::new("Enter the path for the key")
+                        .with_default(&default_path.display().to_string())
+                        .prompt()
+                        .map_err(|e| miette!(e))?
+                        .into();
+
+                    generate_key_file(&path, &key_type, writer)?;
+
+                    PubkeyConfig::Existing(ExistingKeyPath::new(key_type, path))
+                }
+            }
+        }
+        Some(KeyArg::File { path }) => {
+            let path = path.unwrap_or_else(|| {
+                if let Some(parent) = output_path.parent() {
+                    parent.join("homestar.pem")
+                } else {
+                    Settings::path().join("homestar.pem")
+                }
+            });
+
+            generate_key_file(&path, &key_type, writer)?;
+
+            PubkeyConfig::Existing(ExistingKeyPath::new(key_type, path))
+        }
+        Some(KeyArg::Seed { seed: None }) => {
+            let seed = rand::thread_rng().gen::<[u8; 32]>();
+
+            PubkeyConfig::GenerateFromSeed(RNGSeed::new(key_type, seed))
+        }
+        Some(KeyArg::Seed { seed: Some(seed) }) => {
+            let Ok(seed) = PubkeySeed::from_str(&seed) else {
+                bail!("Invalid seed: expected a base64 encoding of 32 bytes")
+            };
+
+            PubkeyConfig::GenerateFromSeed(RNGSeed::new(key_type, seed.0))
+        }
+    };
+
+    config
+        .keypair()
+        .map_err(|e| miette!(format!("Failed to load key: {}", e)))?;
+
+    Ok(config)
+}
+
+fn generate_key_file(
+    path: &PathBuf,
+    key_type: &KeyType,
+    writer: &mut Box<dyn Write>,
+) -> Result<()> {
+    if let Some(parent) = path.parent() {
+        std::fs::create_dir_all(parent).expect("to create parent directory");
+    }
+
+    let key_file = File::options()
+        .read(true)
+        .write(true)
+        .create_new(true)
+        .open(path);
+
+    match key_file {
+        // file did not exist, generate the key
+        Ok(mut file) => {
+            let key = match *key_type {
+                KeyType::Ed25519 => ed25519_compact::KeyPair::generate().sk.to_pem(),
+                KeyType::Secp256k1 => {
+                    std::fs::remove_file(path).expect("to delete key file");
+
+                    bail!("Aborting... generating secp256k1 keys is not yet supported, please provide an existing key file, or choose another key type.")
+                }
+            };
+
+            file.write_all(key.as_bytes())
+                .expect("to write to key file");
+        }
+        // file did exist, do nothing and use existing key
+        Err(err) if err.kind() == std::io::ErrorKind::AlreadyExists => {}
+        err => {
+            err.expect("to open key file");
+        }
+    };
+
+    writeln!(writer, "Writing key file to {:?}", path).expect("to write");
```

Comment:
Hey! Testing out the latest changes. It looks like this message will be logged when using an existing key, passed in through the `--key-file` option. Would it be possible to log something like "Using existing key..." here in this case?

Reply:
Ah whoops, you're totally right. That's what I get for cranking that out while making lunch 😅 Thanks!

Reply:
Fixed in latest push: 
![image](https://github.com/ipvm-wg/homestar/assets/285821/64025143-8c8c-48a9-afdd-c236659e9ed9)

Reply:
Nice! Looks great ❤️